src/actions/indexation/indexable-general-indexation-action.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Reindexation action for indexables.
+ *
+ * @package Yoast\WP\SEO\Actions\Indexation
+ */
+
+namespace Yoast\WP\SEO\Actions\Indexation;
+
+use Yoast\WP\SEO\Builders\Indexable_Builder;
+use Yoast\WP\SEO\Repositories\Indexable_Repository;
+
+/**
+ * Indexable_Misc_Indexation_Action class.
+ */
+class Indexable_General_Indexation_Action implements Indexation_Action_Interface {
+
+	/**
+	 * Represents the indexables repository.
+	 *
+	 * @var Indexable_Repository
+	 */
+	protected $indexable_repository;
+
+	/**
+	 * Represents the indexables builder.
+	 *
+	 * @var Indexable_Builder
+	 */
+	protected $indexable_builder;
+
+	/**
+	 * Indexable_General_Indexation_Action constructor.
+	 *
+	 * @param Indexable_Repository $indexable_repository The indexables repository.
+	 * @param Indexable_Builder    $indexable_builder    The indexables builder.
+	 */
+	public function __construct( Indexable_Repository $indexable_repository, Indexable_Builder $indexable_builder ) {
+		$this->indexable_repository = $indexable_repository;
+		$this->indexable_builder    = $indexable_builder;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_total_unindexed() {
+		$indexables_to_create = $this->query();
+
+		return count( $indexables_to_create );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function index() {
+		$indexables           = [];
+		$indexables_to_create = $this->query();
+
+		if ( \in_array( '404', $indexables_to_create, true ) ) {
+			$indexables[] = $this->indexable_builder->build_for_system_page( '404' );
+		}
+
+		if ( \in_array( 'search', $indexables_to_create, true ) ) {
+			$indexables[] = $this->indexable_builder->build_for_system_page( 'search' );
+		}
+
+		if ( \in_array( 'date_archive', $indexables_to_create, true ) ) {
+			$indexables[] = $this->indexable_builder->build_for_date_archive();
+		}
+
+		if ( \in_array( 'home_page', $indexables_to_create, true ) ) {
+			$indexables[] = $this->indexable_builder->build_for_home_page();
+		}
+
+		return $indexables;
+	}
+
+	/**
+	 * Check which indexables already exists and return the values of the ones to create.
+	 *
+	 * @return array The indexable types to create.
+	 */
+	private function query() {
+		$indexables_to_create = [];
```

Comment:
I'd turn this into a key value array, `[ '404' => true ]`. That would make the above `\in_array` checks a lot cleaner as you can do a simple `isset` if you don't set the keys at all if they don't need to be created.

Or alternatively a boolean check if you do set the keys even if they don't need to be created but set them to false, however that'd make the count more complicated.